src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -1067,4 +1067,8 @@ public void close() throws IOException {
      * @return
      */
     public abstract boolean hasUncommittedChanges();
+
+    public boolean isClosed() {
```

Comment:
I think we should not have this... just call a method and catch EngineCloseException if you wanna test it?